sympy/core/sympify.py

```diff
@@ -138,6 +138,16 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     Traceback (most recent call last):
     ...
     SympifyError: SympifyError: "could not parse 'x***2'"
+    >>> sympify("2x")
+    Traceback (most recent call last):
+    ...
+    SympifyError: SympifyError: "could not parse '2x' "
+
+    sympify function parse the input if it follows the python-syntax.
+    To parse a non-python syntax use "parse_exp":
+    >>> from sympy.parsing.sympy_parser import parse_expr
+    >>> parse_expr("2x", transformations="all")
```

Comment:
This needs to be worded better.

Reply:
I will make the necessary changes.

Reply:
@oscarbenjamin 
I made changes to the documentation, Can you review this again?  : )